include/roboteam_world/kalman/kalmanBall.h

```diff
@@ -11,14 +11,25 @@
 namespace rtt {
 
 class kalmanBall : public kalmanObject {
+    public:
+        enum visState { VISIBLE, EXTRAPOLATED, NOT_VISIBLE };// three states; in first state we actively see the ball.
+        // in EXTRAPOLATED we haven't seen the ball for a while but we do want the KALMAN filter to extrapolate it.
+    private:
+        Vector2 oldVel;
+        visState visibility=NOT_VISIBLE;
     public:
 
         kalmanBall();
 
         //Same as the KalmanObject function but then for ball message
-        roboteam_msgs::WorldBall as_ball_message() const;
+        roboteam_msgs::WorldBall as_ball_message();
         //Same as the KalmanObject function but then for ball frame
+        bool isVisible();
         void kalmanUpdateZ(roboteam_msgs::DetectionBall ball, double timestamp, uint cameraID);
+        void kalmanUpdateX() override;
```

Comment:
why the override?

Reply:
Because the visibility handling for the ball is different (so updateX for the ball is different), so I override the virtual function in kalmanObject.cpp

Reply:
because we use a different updateX() function for visibility reasons.

Reply:
shouldn't then more functions be override?